chain/client/src/stateless_validation/state_witness_tracker.rs

```diff
@@ -0,0 +1,201 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use lru::LruCache;
+use std::hash::Hash;
+
+use crate::metrics;
+use near_async::time::Clock;
+use near_primitives::sharding::ChunkHash;
+use near_primitives::stateless_validation::{ChunkStateWitness, ChunkStateWitnessAck};
+
+/// This is currently used for metrics computation, so we do not need to keep all the
+/// witnesses, so we set some fixed capacity for the witnesses tracked and discard others.
+const CHUNK_STATE_WITNESS_MAX_RECORD_COUNT: usize = 50;
+
+/// Refers to a state witness sent to a chunk producer. Used to locate incoming acknowledgement
+/// messages back to the timing information of the originating witness.
+#[derive(Debug, Clone, Hash, PartialEq, Eq, BorshSerialize, BorshDeserialize)]
+pub struct ChunkStateWitnessKey {
+    /// Hash of the chunk for which the state witness was generated.
+    pub chunk_hash: ChunkHash,
+}
```

Comment:
- Does this struct need to be `pub`?  
- Does `chunk_hash` need to be `pub`?  
- Have you checked if all the derives are absolutely necessary?

Reply:
In general regarding `pub` -- if you can get away by having the item be `pub(crate)` or similar more restricted visibility, then ideally use it for that allows the compiler to spot dead code much more accurately (it can only look at a crate at a time…)

This applies to all new visibility annotations added in this PR.